src/main/scala/org/hammerlab/guacamole/alignment/AffineGapPenaltyAlignment.scala

```diff
@@ -0,0 +1,157 @@
+package org.hammerlab.guacamole.alignment
+
+import breeze.linalg.{ DenseMatrix, argmin }
+import org.hammerlab.guacamole.alignment.AlignmentState.AlignmentState
+
+import scala.annotation.tailrec
+
+object AffineGapPenaltyAlignment {
+
+  /**
+   * Retraces the optimal alignment path from a matrix of scores and transitions
+   *
+   * @param alignmentStates sequenceLength x referenceLength matrix of alignment score at each step
+   * @param alignmentScores sequenceLength x referenceLength matrix of best transition at each step
+   * @param sequenceLength Length of the input sequence to align
+   * @param referenceLength Length of reference sequence to align against
+   * @return An alignment path and score
+   */
+  def findBestPath(alignmentStates: DenseMatrix[AlignmentState],
+                   alignmentScores: DenseMatrix[Double],
+                   sequenceLength: Int,
+                   referenceLength: Int): ReadAlignment = {
+
+    @tailrec
+    def recFindBestPath(sequenceIdx: Int,
+                        referenceIdx: Int,
+                        currentPath: List[AlignmentState]): List[AlignmentState] = {
+
+      val lastMove = alignmentStates(sequenceIdx, referenceIdx)
+      if (sequenceIdx == 1 && referenceIdx == 1) {
+        lastMove :: currentPath
+      } else {
+        lastMove match {
+          case AlignmentState.Match | AlignmentState.Mismatch => {
+            recFindBestPath(sequenceIdx - 1, referenceIdx - 1, lastMove :: currentPath)
+          }
+          case AlignmentState.Insertion => {
+            recFindBestPath(sequenceIdx - 1, referenceIdx, lastMove :: currentPath)
+          }
+          case AlignmentState.Deletion => {
+            recFindBestPath(sequenceIdx, referenceIdx - 1, lastMove :: currentPath)
+          }
+        }
+      }
+    }
+
+    val alignments = recFindBestPath(sequenceLength, referenceLength, List.empty)
+    ReadAlignment(alignments, argmin(alignmentScores.t(::, sequenceLength)))
+  }
+
+  /**
+   * Produces an alignment of an input sequence against a reference sequence
+   *
+   * @param sequence Input sequence to align
+   * @param reference Reference sequence to align against
+   * @param mismatchProbability Penalty to having a mismatch in the alignment
+   * @param openGapProbability Penalty to start an insertion or deletion in the alignment
+   * @param closeGapProbability Penalty to end in an insertion or deletion in the alignment
+   * @return An alignment path and score
+   */
+  def align(sequence: Seq[Byte],
+            reference: Seq[Byte],
+            mismatchProbability: Double = 1e-3,
```

Comment:
Not particularly, and they do need more experimentation.
Right now these result in reasonable alignments, but for example BWA sets these as 
mismatchProbability = e^-4 ~=0.018
gapOpen = e^-6 ~=0.0024
gapExtends= e^-1 ~=0.36

Plugging those in actually does give the same alignments as well, but seemed a bit more arbitrary.